send/slack

```diff
@@ -0,0 +1,518 @@
+
+
+def get_their_channels(webclient):
+	print_debug("Fetching Slack channels")
+	channels = []
+	cursor = None
+	while cursor != "":
+		response = webclient.conversations_list(
+			cursor=cursor,
+			exclude_archived=False,
+			limit=COUNT,
+			types="public_channel,private_channel",
+		)
+		check_webclient_response(response)
+		channels.extend(response["channels"])
+		cursor = response["response_metadata"].get("next_cursor")
+
+	return channels
+
+
+def get_their_users_web(webclient):
+	print_debug("Fetching Slack users")
+	users = []
+	cursor = None
+	while cursor != "":
+		response = webclient.users_list(cursor=cursor, limit=COUNT)
+		check_webclient_response(response)
+		users.extend(response["members"])
+		cursor = response["response_metadata"].get("next_cursor")
+	return users
+
+
+def get_their_members(webclient, channel_id):
+	print_debug(f"Fetching members of channel {channel_id}")
+	members = []
+	cursor = None
+	while cursor != "":
+		response = webclient.conversations_members(
+			cursor=cursor,
+			channel=channel_id,
+			limit=COUNT,
+		)
+		check_webclient_response(response)
+		members.extend(response["members"])
+		cursor = response["response_metadata"].get("next_cursor")
+
+	return members
+
+
+def invite_members(webclient, members, channel_id):
+	print_debug(f"Inviting user {members} to {channel_id}")
+	response = webclient.conversations_invite(channel=channel_id, users=members)
+	if response.get("ok", False) and response.get("error", "").startswith(
+		"already_in_channel"
+	):
+		if len(members) == 1:
+			print_debug(
+				f"Tried to invite existing member {members} of channel {channel_id}, skipped."
+			)
+			return
+		else:
+			for member in members:
+				invite_members(webclient, [member], channel_id)
+
+	check_webclient_response(response)
+	global MEMBERS_ADDED
+	MEMBERS_ADDED += len(members)
+
+
+def kick_members(webclient, members, channel_id):
+	for user in members:
+		print_debug(f"Kicking user {user} from channel {channel_id}")
+		response = webclient.conversations_kick(channel=channel_id, user=user)
+		check_webclient_response(response)
+		global MEMBERS_REMOVED
+		MEMBERS_REMOVED += len(members)
+
+
+def archive_channel(webclient, channel_id):
+	print_debug(f"Archiving channel {channel_id}")
+	response = webclient.conversations_archive(channel=channel_id)
+	check_webclient_response(response)
+	global CHANNELS_ARCHIVED
+	CHANNELS_ARCHIVED += 1
+
+
+def unarchive_channel(webclient, channel_id):
+	print_debug(f"Restoring archivated channel {channel_id}")
+	response = webclient.conversations_unarchive(channel=channel_id)
+	check_webclient_response(response)
+	global CHANNELS_UPDATED
+	CHANNELS_UPDATED += 1
+
+
+def notify_users_deactivated(user_ids):
+	print_debug(f"Pushing notification about {len(user_ids)} deactivated users")
+	formatted_ids = ", ".join(["<@" + u + ">" for u in user_ids])
+	text = f"[UPOZORNĚNÍ] : Došlo k deaktivaci {'uživatelů' if len(user_ids) > 1 else 'uživatele'} {formatted_ids}.\n"
```

Comment:
I suppose this should be in czech?

Reply:
for now it is sufficient. in case someone else would like to use it, we can make translations and set some language facility attribute